hyperspy/io_plugins/digital_micrograph.py

```diff
@@ -785,11 +786,11 @@ def _get_mode(self, mode):
             return 'TEM'
 
     def _get_time(self, time):
-        dt = datetime.strptime(time, "%I:%M:%S %p")
+        dt = dateutil.parser.parse(time)
         return dt.time().isoformat()
```

Comment:
This solves the locale issue, but it will still throw an either `OverflowError` or `ValueError` if the `time` variable is wrong. So I think the parser should be in a `try` `except`, in case the `time` variable is something really weird. For example:
```Python
try:
    dt = dateutil.parser.parse(time)
    return dt.time().isoformat()
except:
    _logger.warning("Time string, %s,  could not be parsed", time)
    return("") # Or whatever the default value should be
    return("") # T

If the